padrino-core/lib/padrino-core/application/routing.rb

```diff
@@ -21,6 +21,10 @@ module Routing
     # Raised when a route was invalid or cannot be processed.
     class UnrecognizedException < RuntimeError; end
 
+    # Raised when block arity was nonzero and was not same with
+    # captured parameter length.
+    class WrongArityException < RuntimeError; end
```

Comment:
I would make this a subclass of ArgumentError, which is a `StandardError`, though.

In general, wouldn't `StandardError` be the better pick anyways?